Server PHP/classes.php

```diff
@@ -46,16 +51,117 @@ function setSocket($sock) {
   function setEmail($usrEmail) {
     $this->email = $usrEmail;
   }
+  function setNumFavorite($num) {
+    $this->numFavorite = $num;
+  }
+  function setNumRecent($num) {
+    $this->numRecent = $num;
+  }
   function setGroup($groupID) {
-    array_unshift($this->group, $groupID);
+    array_unshift($this->currGroups, $groupID);
+  }
+  function setRecGroup($groupID) {
+    if (in_array($groupID, $this->recGroups)) {
+      $key = array_search($groupID, $this->recGroups);
+      if ($key != 0){
+        $count = count($this->recGroups);
+        echo "DEBUG: Trying to add to Recent Group this is the count before add: $count\n";
+        if(count($this->recGroups) >= $this->numRecent){
+          array_unshift($this->recGroups, $groupID);
+          $this->recGroups = array_splice($this->recGroups, 0, $this->numRecent);
+        }
+        else
+          array_unshift($this->recGroups, $groupID);
+      }
+    }//if in array
+    else {
+      if(count($this->recGroups) >= $this->numRecent) {
+        array_unshift($this->recGroups, $groupID);
+        $this->recGroups = array_splice($this->recGroups, 0, 5);
+      }
+      else
+        array_unshift($this->recGroups, $groupID);
+    }
+  }
+  function setFavGroup($groupID) {
+    if (count($this->favGroups) >= $this->numFavorite) {
+      array_unshift($this->favGroups, $groupID);
+      $this->recGroups = array_splice($this->recGroups, 0, $this->numFavorite);
+    }
+    else
+      array_unshift($this->favGroups, $groupID);
+  }
+  function removeFavGroup($groupID) {
+    if (in_array($groupID, $this->favGroups)) {
+      $key = array_search($groupID, $this->favGroups);
+      array_splice($this->favGroups, $key, 1);
+    }
   }
   function removeGroup($groupID) {
-    for($i = 0; $i < (count($this->group)); $i++) {
-      if($this->group[$i] == $groupID)
-        array_splice($this->group, $i, 1);
+    if (in_array($groupID, $this->currGroups)) {
+      $key = array_search($groupID, $this->currGroups);
+      array_splice($this->currGroups, $key, 1);
     }
   }
+  function clearGroups() {
+    $this->recGroups = array();
+    $this->currGroups = array();
+    $this->favGroups = array();
+  }
+}
 
+class Group {
+  function _construct() {
+    echo "Creating Group Class\n";
+  }
+  function __destruct() {
+    echo "Destructing Group Class\n";
+  }
+
+  var $groupID;
+  var $numMembers;
+  var $admin;
+  var $members = array();
+  var $memberIPs = array();
+
+  //Get Functions
+  function getGroupID() {
+    return $this->groupID;
+  }
+  function getNumMembers() {
+    $this->numMembers = count($this->members);
+    return $this->numMembers;
+  }
+  function getMembers() {
+    return $this->members;
+  }
+  function getMemberIP() {
+    return $this->memberIPs;
+  }
+  function getAdmin() {
+    return $this->admin;
+  }
+  //Set Functions
+  function setGroupID($groupname) {
+      $this->groupID = $groupname;
+  }
+  function setMember($member) {

```

Comment:
I think renaming to addMember would be clearer, and should this work with the user objects so it adds not just the name but the actual user?